BigBaseV2/src/memory/pattern.cpp

```diff
@@ -3,67 +3,48 @@
 
 namespace memory
 {
-	pattern::pattern(std::string_view ida_sig)
+	std::optional<std::uint8_t> to_hex(char const c)
 	{
-		auto to_upper = [](char c) -> char
-		{
-			return c >= 'a' && c <= 'z' ? static_cast<char>(c + ('A' - 'a')) : static_cast<char>(c);
-		};
-
-		auto to_hex = [&](char c) -> std::optional<std::uint8_t>
+		switch (c)
 		{
-			switch (to_upper(c))
-			{
-			case '0':
-				return static_cast<std::uint8_t>(0);
-			case '1':
-				return static_cast<std::uint8_t>(1);
-			case '2':
-				return static_cast<std::uint8_t>(2);
-			case '3':
-				return static_cast<std::uint8_t>(3);
-			case '4':
-				return static_cast<std::uint8_t>(4);
-			case '5':
-				return static_cast<std::uint8_t>(5);
-			case '6':
-				return static_cast<std::uint8_t>(6);
-			case '7':
-				return static_cast<std::uint8_t>(7);
-			case '8':
-				return static_cast<std::uint8_t>(8);
-			case '9':
-				return static_cast<std::uint8_t>(9);
-			case 'A':
-				return static_cast<std::uint8_t>(10);
-			case 'B':
-				return static_cast<std::uint8_t>(11);
-			case 'C':
-				return static_cast<std::uint8_t>(12);
-			case 'D':
-				return static_cast<std::uint8_t>(13);
-			case 'E':
-				return static_cast<std::uint8_t>(14);
-			case 'F':
-				return static_cast<std::uint8_t>(15);
-			default:
-				return std::nullopt;
-			}
-		};
-
-		for (std::size_t i = 0; i < ida_sig.size(); ++i)
+		case '0': return static_cast<std::uint8_t>(0x0);
+		case '1': return static_cast<std::uint8_t>(0x1);
+		case '2': return static_cast<std::uint8_t>(0x2);
+		case '3': return static_cast<std::uint8_t>(0x3);
+		case '4': return static_cast<std::uint8_t>(0x4);
+		case '5': return static_cast<std::uint8_t>(0x5);
+		case '6': return static_cast<std::uint8_t>(0x6);
+		case '7': return static_cast<std::uint8_t>(0x7);
+		case '8': return static_cast<std::uint8_t>(0x8);
+		case '9': return static_cast<std::uint8_t>(0x9);
+		case 'a': return static_cast<std::uint8_t>(0xa);
+		case 'b': return static_cast<std::uint8_t>(0xb);
+		case 'c': return static_cast<std::uint8_t>(0xc);
+		case 'd': return static_cast<std::uint8_t>(0xd);
+		case 'e': return static_cast<std::uint8_t>(0xe);
+		case 'f': return static_cast<std::uint8_t>(0xf);
+		case 'A': return static_cast<std::uint8_t>(0xA);
+		case 'B': return static_cast<std::uint8_t>(0xB);
+		case 'C': return static_cast<std::uint8_t>(0xC);
+		case 'D': return static_cast<std::uint8_t>(0xD);
+		case 'E': return static_cast<std::uint8_t>(0xE);
+		case 'F': return static_cast<std::uint8_t>(0xF);
+		default: return std::nullopt;
+		}
+	}
+	pattern::pattern(std::string_view ida_sig)
+	{
+		for (std::size_t i{}; i != ida_sig.size(); ++i)
```

Comment:
Does the compiler optimise this ida_sig.size() call or does it call .size() each iteration?

Reply:
It should, let me change

Reply:
Committed requested changes.